robottests/test_remotescript.robot

```diff
@@ -2,31 +2,241 @@
+    ${ret2}=    Copy File To Target  source_file=${CURDIR}/target_script.sh  destination_dir=/tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0
+    ${ret3}=     Execute Script In Target    file=/tmp/target_dir/target_script.sh
+    Should Be Equal As Integers     ${ret3.status}  0
+    [Teardown]  Execute Command In Target  rm -r /tmp/target_dir
+
+RemoteScript Execute Background Command In Target
+	FOR		${exec_id}	IN	bg1  bg2  bg3
+			Execute Background Command In Target	command=echo Hello;echo World!>&2
+			...										target=default exec_id=${exec_id}
+	END
+	FOR		${exec_id}	IN	bg1  bg2  bg3
+		${result}=	Wait Background Execution	${exec_id}
+		Should Be Equal As Integers		${result.status}	0
+		Should Be Equal		${result.stdout}	Hello
+		Should Be Equal		${result.stderr}	World!
+	END
+
+RemoteScript Execute Background Script In Target
+    Better Create Directory In Target
+    ${out}=     Run Process  ls ${CURDIR}  shell=${True}
+    Log  ${out.stdout}
+    ${ret2}=    Copy File To Target  source_file=${CURDIR}/target_script.sh  destination_dir=/tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0
+    Execute Background Script In Target  file=/tmp/target_dir/target_script.sh  target=default  exec_id=bg1
+    ${result_bg1}=  Wait Background Execution   exec_id=bg1
+    Log  ${result_bg1}
+    Should Be Equal As Integers     ${result_bg1.status}   0
+    [Teardown]  Execute Command In Target   rm -r /tmp/target_dir
+
+RemoteScript Copy File Between Targets
+    RemoteScript Set Target With Sshkeyfile
+    Better Create Directory In Target
+    Better Create Directory In Target  target=host2
+    Create File In Target
+    ${ret}=     Copy File Between Targets
+    ...         from_target=default
+    ...         source_file=/tmp/default/${TARGET_FILE}
+    ...         to_target=host2
+    ...         destination_dir=/tmp/host2
+    Should Be Equal As Integers     ${ret.status}   0
+    Log  ${ret}
+
+RemoteScript Kill Background Execution
+    Execute Background Command In Target    command=echo Hello; sleep 3   target=default  exec_id=bg1
+    Sleep  1
+    Kill Background Execution  exec_id=bg1
+    ${ret}=     Wait Background Execution  exec_id=bg1
+    Log  ${ret}
+
+RemoteScript Set Target Property
+    RemoteScript Set Target With Sshkeyfile
+    ${prop}=    Get Target Properties  target=default
+    FOR     ${key}  IN  @{prop.keys()}
+        Set Target Property  target_name=host2  property_name=${key}  property_value=${prop["${key}"]}
+    END
+    ${tprop}=   Get Target Properties  target=host2
+    Log  ${prop}
+    Log  ${tprop}
+    Dictionary Should Contain Sub Dictionary  ${prop}  ${tprop}
+
+RemoteScript Set Default Target Property
+    RemoteScript Set Target With Sshkeyfile
+    ${prop}=    Get Target Properties   target=default
+    FOR     ${key}  IN  @{prop.keys()}
+        Set Default Target Property  property_name=${key}    property_value=${prop["${key}"]}
+    END
```

Comment:
Please change default target properties from their defaults in order to avoid false positive results.